qiskit_alice_bob_provider/plugins/state_preparation.py

```diff
@@ -198,6 +207,84 @@ def run(self, dag: DAGCircuit) -> DAGCircuit:
         return dag
 
 
+class CustomUnitarySynthesis(UnitarySynthesis):
+    """
+    Synthesize gates according to their basis gates.
+
+    This is a replacement of the base UnitarySynthesis pass to handle the
+    SK synthesis on Alice & Bob targets.
+
+    The pass overrides the _run_main_loop method to remove an "optimization"
+    update that skips non control operational nodes, such as the gates we
+    synthesize with Solovay-Kitaev (rx, rz...)
+    https://github.com/Qiskit/qiskit/commit/a02e0a3dc6efcbe69cc86718be9b527c73367395
```

Comment:
Apparently it's this commit: d2ab4dfb

Also, let's keep open an issue on qiskit to see if they can fix it on their side (so that we don't have to maitain our own CustomUnitarySynthesis class) or if there is a better way to do what we need to do !